synapse/util/__init__.py

```diff
@@ -78,22 +80,22 @@ class Clock:
     _reactor = attr.ib()
 
     @defer.inlineCallbacks
-    def sleep(self, seconds):
-        d = defer.Deferred()
+    def sleep(self, seconds: float):
+        d: defer.Deferred[float] = defer.Deferred()
         with context.PreserveLoggingContext():
             self._reactor.callLater(seconds, d.callback, seconds)
             res = yield d
         return res
 
-    def time(self):
+    def time(self) -> float:
         """Returns the current system time in seconds since epoch."""
         return self._reactor.seconds()
 
-    def time_msec(self):
+    def time_msec(self) -> int:
         """Returns the current system time in milliseconds since epoch."""
         return int(self.time() * 1000)
 
-    def looping_call(self, f, msec, *args, **kwargs):
+    def looping_call(self, f, msec, *args, **kwargs) -> LoopingCall:
```

Comment:
```suggestion
    def looping_call(self, f: Callable, msec: int, *args, **kwargs) -> LoopingCall:
```

I think its worth at least giving some of the types here. When 3.10 is released we might be able to use `ParamSpec` for the function bounds here.